### ISSUE TITLE
IMP

### ISSUE THREAD

**Issue by larsbrinkhoff** [charlesUnixPro/dps8m#4]
Hello,

If you consider adding some IMP functionality, I'd like to be involved peripherally.

I have ITS using an IMP interface as part of a KA10 simulator for SIMH.  And also a work in progress to emulate the IMP software in a standalone C program.

**Comment by charlesUnixPro** [comment]
On Fri, Oct 5, 2018 at 12:03 AM Lars Brinkhoff <notifications@github.com>
wrote:

> Hello,
>
> If you consider adding some IMP functionality, I'd like to be involved
> peripherally.
>
> I have ITS using an IMP interface as part of a KA10 simulator for SIMH.
> And also a work in progress to emulate the IMP software in a standalone C
> program.
>
>
The IMP code uses udplib for IMP-to-IMP communications; I added code to use
udplib for IMP-to-host communication.

My goal was to restore ARPAnet functionality to Multics by adding the
"ABSI" device to the dps8m emulator, with partial success. (I have not
implemented the "ready" bits; because udplib is connectionless, some kind
of heartbeat is needed tithe the ready bits wired to a heartbeat timeout,
or some similar scheme.)

It turns out that too much of the Multics ARPAnet software is missing; work
has started on recovering some backup tapes that may contain the missing
pieces. (It was an unbundled product, as was TCP/IP support, so currently
Multics does not support any network connectivity other then  modem
emulation with TELNET.

So at this point IMP support is on hiatus pending reconstruction of missing
Multics software.

-- Charles

**Comment by larsbrinkhoff** [comment]
Yes, I understand this is some distance into the future.  Just a heads up.

In my vauge reasoning, I was thinking the IMP-to-IMP dedicated line might have been implemented as a TCP connection.  But UDP would work too.  Any particular reason you chose one over the other?
